### PR TITLE
Backend/feature/refactor routes

### PR DESCRIPTION
**Pull request by Dhamareshwarakumar**
updated category and product routes

### CODE REVIEW

**Review by sunkarivasu — APPROVED** [review]
great work @Dhamareshwarakumar

File: routes/categories.js [hunk 1]
```diff
@@ -1,75 +1,109 @@
...
+        .catch((err) => { res.status(400).json("Error while fetching categories:" + err) });
 });
 
-router.route("/updateCategoryDetails/").put((req,res) =>
-{
-    Category.updateOne({categoryName:req.body.category},{$set:{subCategoryList:req.body.subCategoryList,categoryImage:req.body.categoryImage}})
-    .then(()=>{res.json("Category updated successfully")})
-    .catch(()=>{res.status(400).json("Error:"+err)})
+router.route("/updateCategoryDetails/").put((req, res) => {
+    Category.updateOne({ categoryName: req.body.category }, { $set: { subCategoryList: req.body.subCategoryList, categoryImage: req.body.categoryImage } })
```
Comment (AjayVenkatesh2026) [inline routes/categories.js:83]: is there a change where either `req.body.subCategoryList` or `req.body.subCategoryList` be undefined ? and if so, can it lead to unpredicatable behaviour for this api ?

**Review by AjayVenkatesh2026 — COMMENTED** [review]
seems like @sunkarivasu already merged the MR.

File: routes/categories.js (hunk 1 above)
Comment (sunkarivasu) [reply routes/categories.js:83]: router.route("/updateCategoryDetails/").put((req, res) => {}

this route will be modified by @Dhamareshwarakumar
he modified only few endpoints in this pr

Comment (AjayVenkatesh2026) [reply routes/categories.js:83]: got it

**Pull request merged**